aiosmtpd/controller.py

```diff
@@ -8,13 +8,16 @@
 
 @public
 class Controller:
-    def __init__(self, handler, loop=None, hostname='::0', port=8025):
+    def __init__(self, handler, loop=None, hostname='::0', port=8025,
+                 thread_timeout=None):
```

Comment:
I think it makes sense to make it timeout by default so that errors could be caught, but didn't want to change the original behavior right away.

Since the ready event was initially added to wait for the server to complete it's setup a very short timeout value might cause it to break early. Since you know the ins and outs of the SMTP server implementation you could probably estimate a better timeout value than me. However, the way I'm running it right now a 1 second timeout seem to be more than enough. One thing to consider though is the fact that the factory is overridable which could cause confusion for people implementing an SMTP server that have a longer setup.

I'll update it with a default timeout of 5 seconds to be conservative but also not making the user wait too long. Let me know if you want to change it.